kernel/src/main/scala-2.13+/cats/kernel/instances/LazyListInstances.scala

```diff
@@ -73,3 +77,9 @@ class LazyListMonoid[A] extends Monoid[LazyList[A]] {
   override def combineAll(xs: IterableOnce[LazyList[A]]): LazyList[A] =
     StaticMethods.combineAllIterable(LazyList.newBuilder[A], xs)
 }
+
+object LazyListMonoid {
+  @nowarn("msg=deprecated")
+  private[this] val singleton: Monoid[LazyList[Any]] = new LazyListMonoid[Any]
```

Comment:
Perhaps, in the light of Scala3 and future migrations we should avoid using `private [this]` modifier:
https://docs.scala-lang.org/scala3/reference/dropped-features/this-qualifier.html

I think just `private` would be private enough :)

Reply:
Just for the record, here's the difference.

```scala
//> using scala "2.13.8"

object Foo {
  private val bar: String = "bar"
  private[this] val baz: String = "baz"

  def publicBar = bar
  def publicBaz = baz
}
```

```java
/*
 * Decompiled with CFR 0.152.
 */
public final class Foo$ {
    public static final Foo$ MODULE$ = new Foo$();
    private static final String bar = "bar";
    private static final String baz = "baz";

    private String bar() {
        return bar;
    }

    public String publicBar() {
        return this.bar();
    }

    public String publicBaz() {
        return baz;
    }

    private Foo$() {
    }
}
```

Reply:
Yeah, when scala 3 actually deprecates/removes it we can replace `private[this]` with `private` but there seems no harm using it in the meantime

Reply:
I'm totally fine with `private[this]` too, I don't think it is that important for now.

---

I personally prefer to avoid using `private [this]` on project that are supposed to be migrated to Scala3 eventually. But since the majority of projects are still using Scala2, they may benefit from such optimization.

Btw, I've just checked the @armanbilge's example with Scala 3.2.0 (just out of curiosity :) ) and looks like Scala3 does a great job on optimizing `private` fields indeed:

<details>

```java
/*
 * Decompiled with CFR 0.152.
 * 
 * Could not load the following classes:
 *  scala.runtime.ModuleSerializationProxy
 */
import java.io.Serializable;
import scala.runtime.ModuleSerializationProxy;

public final class Foo$
implements Serializable {
    private static final String bar;
    private static final String baz;
    public static final Foo$ MODULE$;

    private Foo$() {
    }

    static {
        MODULE$ = new Foo$();
        bar = "bar";
        baz = "baz";
    }

    private Object writeReplace() {
        return new ModuleSerializationProxy(Foo$.class);
    }

    public String publicBar() {
        return bar;
    }

    public String publicBaz() {
        return baz;
    }
}
```

</details>

Reply:
> I personally prefer to avoid using private [this] on project that are supposed to be migrated to Scala3 eventually.

I am 💯 % confident that find/replace `private[this]` to `private` will be the fastest and easiest part of Scala 3 migration in Cat :)